src/include/duckdb/parser/constraints/generated_constraint.hpp

```diff
@@ -0,0 +1,36 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/parser/constraints/unique_constraint.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "duckdb/parser/constraint.hpp"
+#include "duckdb/common/vector.hpp"
+
+namespace duckdb {
+
+class GeneratedConstraint : public Constraint {
+public:
+	DUCKDB_API GeneratedConstraint(uint64_t index, unique_ptr<ParsedExpression> expression);
```

Comment:
This is only used in the transformer, and not in the main system, right? Perhaps we could remove this class entirely and do a check prior to calling `TransformConstraint` in `transform_create_table.cpp`. That way we would not need to create an entirely new constraint class there.